lib/protocol/protocol.go

```diff
@@ -804,13 +804,13 @@ func (c *rawConnection) writeCompressedMessage(msg message, marshaled []byte, ov
 	}
 
 	cOverhead := 2 + hdrSize + 4
-	maxCompressed := cOverhead + lz4.CompressBlockBound(len(marshaled))
+	maxCompressed := cOverhead + len(marshaled) - len(marshaled)/32
```

Comment:
Comment to address the reason for the somewhat magical subtraction and division by 32 etc here I think would be nice

Reply:
Done. Also added a tiny additional optimization.